lucille-plugins/lucille-weaviate/src/main/java/com/kmwllc/lucille/weaviate/stage/DropIndexedDocuments.java

```diff
@@ -0,0 +1,304 @@
+
+        // add ids to set
+        ids.addAll(batchIds);
+
+        // make a string of all the ids and write them to a file
+        StringBuilder sb = new StringBuilder();
+        for (String id : batchIds) {
+          sb.append(id).append("\n");
+        }
+        writer.write(sb.toString());
+        writer.flush();
+
+        // update counter and number left to index
+        count++;
+        numIndexed -= batchSize;
+
+        log.info("Processed batch {} of {}", count, totalBatches);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    log.info("Wrote IDs to " + fileWithIds);
+    return ids;
+  }
+
+  private Pair<List<String>, String> getBatchIds(String lastID) throws StageException {
+
+    // initialize fields to get from Weaviate
+    Field[] fields = new Field[]{
+        Field.builder()
+            .name(idDestinationName)
+            .build(),
+        Field.builder()
+            .name("_additional")
+            .fields(
+                Field.builder()
+                    .name("id")
+                    .build()
+            ).build()
+    };
+
+    // initialize a request
+    Get query = client
+        .graphQL()
+        .get()
+        .withClassName(weaviateClassName)
+        .withFields(fields)
+        .withLimit(batchSize);
+
+    // if lastID is not null, add it to the request for pagination
+    if (lastID != null) {
+      query.withAfter(lastID);
+    }
+
+    // run query
+    Result<GraphQLResponse> response = query.run();
+
+    if (response.hasErrors()) {
+      throw new StageException("Error getting ids from Weaviate: " + response.getError().toString());
+    }
+
+    JsonArray array = GSON.toJsonTree(response.getResult().getData())
+        .getAsJsonObject()
+        .get("Get")
+        .getAsJsonObject()
+        .get(weaviateClassName)
+        .getAsJsonArray();
+
+    List<String> parsedIDs = new ArrayList<>();
+    for (int i = 0; i < array.size(); i++) {
+      JsonObject obj = array.get(i).getAsJsonObject();
+      JsonObject additional = obj.get("_additional").getAsJsonObject();
+
+      parsedIDs.add(obj.get("id_original").getAsString());
+
+      if (i == array.size() - 1) {
+        lastID = additional.get("id").getAsString();
+      }
+    }
+
+    return new Pair<>(parsedIDs, lastID);
+  }
+
+  private int getCount() throws StageException {
+
+    Field fields = Field.builder()
+        .name("meta")
+        .fields(
+            Field.builder()
+                .name("count")
+                .build()
+        ).build();
+
+    Result<GraphQLResponse> response = client
+        .graphQL()
+        .aggregate()
+        .withClassName(weaviateClassName)
+        .withFields(fields)
+        .run();
+
+    if (response.hasErrors()) {
+      throw new StageException("Error getting count from Weaviate: " + response.getError().toString());
+    }
+
+    JsonArray array = GSON.toJsonTree(response.getResult().getData())
```

Comment:
We use Jackson in Lucille for json manipulation, not GSON. 
Can leave for now but we should have a TODO to standardize this code so it uses Jackson.